alexwind/main.js

```diff
@@ -0,0 +1,104 @@
+let emptyDoneList = true;
+
+const toDo = {
+    todoList : {
+        vacuumCleaning : 'to Do',
+        washDishes     : 'Done',
+        TakeOutTrash   : 'in Progress', 
+        KickAss        : 'Done',       
+    },    
+};
+
+const Status = {
+    TODO : 'to Do',
+    INPROGRESS : 'in Progress' ,    
+    DONE : 'Done',   
+};
+
+function addTask(newTask, newStatus) { // добавление задачи.
+    toDo.todoList[newTask] = newStatus;
+};
+
+function changeStatus(nameTask, newStatus) { // смена статуса.
+    if((typeof nameTask === 'string' && typeof newStatus === 'string')) {
+         return toDo.todoList[nameTask] = newStatus;
+    } else {
+        return 'Incorrect data type : expected string';
+    };
+};
+
+function deleteTask(name) { // удаление задачи.
+    delete toDo.todoList[name];
+};
+
+function showList() { //отображение списка задач.
+    for (const keyTask in toDo.todoList) {
+        if(toDo.todoList[keyTask] === Status.TODO) {
+            console.log('"' + keyTask + '" : ' + toDo.todoList[keyTask]);
+        };
+    };
+
+    for (const keyTask in toDo.todoList) {
+        if(toDo.todoList[keyTask] === Status.INPROGRESS) {
+            console.log('"' + keyTask + '" : ' + toDo.todoList[keyTask]);
+        };
+    };
+
+    for (const keyTask in toDo.todoList) {
+        if(toDo.todoList[keyTask] === Status.DONE) {
+            console.log('"' + keyTask + '" : ' + toDo.todoList[keyTask]);
+            emptyDoneList = false;
+        };
+    };
+  
+    if(emptyDoneList) {
+        console.log('Nothing in done')
+    }
+};
+
+let emptyList = true;
+
+function showListHard() { // отображение лист задач ( задача со звездочкой )
+    for (const keyStatus in Status) {
+        console.log(Status[keyStatus] + ':');
+
+        for (const keyTask in toDo.todoList) {
+            if(toDo.todoList[keyTask] === Status.TODO && Status.TODO === Status[keyStatus]) {
```

Comment:
Это условие можно брать из внешнего цикла и избавиться от повторения кода
```suggestion
            if(toDo.todoList[keyTask] === Status[keyStatus]) {
```